internal/providers/nscloud/api/rpc.go

```diff
@@ -102,6 +101,31 @@ func MakeAPI(endpoint string) API {
 	}
 }
 
+func FetchTenantToken(ctx context.Context) (*auth.Token, error) {
+	return tasks.Return(ctx, tasks.Action("nscloud.fetch-tenant-token"), func(ctx context.Context) (*auth.Token, error) {
+		if !fnapi.AdminMode {
+			t, err := auth.LoadTenantToken(ctx)
+			if err != nil {
+				return nil, err
+			}
+			return t, nil
+		}
+
+		// In admin mode we exchange user token to a tenant token with `admin` scope.
+		userToken, err := auth.GenerateToken(ctx)
+		if err != nil {
+			return nil, err
+		}
+
+		t, err := fnapi.ExchangeUserToken(ctx, userToken, []string{"admin"})
```

Comment:
Let's change `ExchangeUserToken` to receive scopes as `...string` and add a string constant `AdminScope`

Reply:
True, looks better. Done.